indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,310 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import timeTableURL from 'indico-url:event_registration.api_session_blocks';
+
+import _ from 'lodash';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+import {Checkbox, Form, Accordion, AccordionTitle, AccordionContent, Icon} from 'semantic-ui-react';
+
+import {FinalDropdown, FinalField, FinalInput, validators as v} from 'indico/react/forms';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate, Param, Plural, PluralTranslate, Singular} from 'indico/react/i18n';
+
+import {getStaticData} from '../selectors';
+
+import '../../../styles/regform.module.scss';
+
+const sessionDataSchema = PropTypes.arrayOf(
+  PropTypes.shape({
+    fullTitle: PropTypes.string.isRequired,
+    id: PropTypes.number.isRequired,
+    startDt: PropTypes.string.isRequired,
+  })
+);
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  onFocus,
+  onBlur,
+  display,
+  minimum,
+  maximum,
+  ...props
+}) {
+  const [expandedHeaders, setExpandedHeaders] = useState([]);
+
+  const markTouched = () => {
+    onFocus();
+    onBlur();
+  };
+
+  function handleValueChange(e, data) {
+    markTouched();
+    const id = data.value;
+    const newValue = _.xor(value, [id]);
+    onChange(newValue);
+  }
+
+  function handleHeaderClick(e, accordion) {
+    if (expandedHeaders.includes(accordion.index)) {
+      setExpandedHeaders(expandedHeaders.filter(el => el !== accordion.index));
+    } else {
+      setExpandedHeaders([...expandedHeaders, accordion.index]);
+    }
+  }
+
+  useEffect(() => {
+    setExpandedHeaders(
+      sessionData && display === 'expanded' ? [...Array(_.keys(sessionData).length).keys()] : []
```

Comment:
The middle part could be simplified to `_.range(Object.keys(sessionData).length)`.

However, I don't understand why we use indexes here. Why not use an object instead that the dates as keys?

```js
    setExpandedHeaders(Object.fromEntries(Object.keys(sessionData || {}).map(date => [date, display === 'expanded'])))
```

This would also allow making handleHeaderClick much more simple. You'd just need to pass `onClick={handleHeaderClick(date)}` below.

```js
  const handleHeaderClick = date => () => {
    setExpandedHeaders({...expandedHeaders, [date]: !expandedHeaders[date]});
  }
```